src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -0,0 +1,153 @@
+    typeTable.saveNicknameFor("java.util.List");
+    typeTable.saveNicknameFor("com.google.api.gax.testing.MockServiceHelper");
+    typeTable.saveNicknameFor("com.google.api.gax.testing.MockGrpcService");
+    typeTable.saveNicknameFor("com.google.protobuf.GeneratedMessage");
+    typeTable.saveNicknameFor("junit.framework.Assert");
+  }
+
+  private GapicSurfaceTestClassView createTestClassView(SurfaceTransformerContext context) {
+    Interface service = context.getInterface();
+    String outputPath = pathMapper.getOutputPath(service, context.getApiConfig());
+
+    GapicSurfaceTestClassView testClass =
+        GapicSurfaceTestClassView.newBuilder()
+            .packageName(context.getApiConfig().getPackageName())
+            .apiSettingsClassName(context.getNamer().getApiSettingsClassName(service))
+            .apiClassName(context.getNamer().getApiWrapperClassName(service))
+            .name(context.getNamer().getTestClassName(service))
+            .mockServiceClassName(context.getNamer().getMockServiceClassName(service))
+            .testCases(createTestCaseViews(context))
+            .outputPath(outputPath)
+            .imports(context.getTypeTable().getImports())
+            .templateFileName(TEST_TEMPLATE_FILE)
+            .build();
+    return testClass;
+  }
+
+  private List<GapicSurfaceTestCaseView> createTestCaseViews(SurfaceTransformerContext context) {
+    ArrayList<GapicSurfaceTestCaseView> testCaseViews = new ArrayList<>();
+    for (Method method : context.getInterface().getMethods()) {
+      MethodTransformerContext methodContext = context.asMethodContext(method);
+      testCaseViews.add(createTestCaseView(methodContext));
+    }
+    return testCaseViews;
+  }
+
+  private GapicSurfaceTestCaseView createTestCaseView(MethodTransformerContext methodContext) {
+    MethodConfig methodConfig = methodContext.getMethodConfig();
+    InitCodeTransformer initCodeTransformer = new InitCodeTransformer();
+    InitCodeView initCodeView =
+        initCodeTransformer.generateInitCode(methodContext, methodConfig.getRequiredFields());
+    List<GapicSurfaceTestAssertView> assertViews =
+        initCodeTransformer.generateTestAssertViews(
+            methodContext, methodConfig.getRequiredFields());
+
+    String resourceTypeName = "";
+    boolean isPageStreaming = methodConfig.isPageStreaming();
+    if (isPageStreaming) {
+      resourceTypeName =
+          methodContext
+              .getTypeTable()
+              .getAndSaveNicknameForElementType(
+                  methodConfig.getPageStreaming().getResourcesField().getType());
+    }
+
+    String requestTypeName =
+        methodContext
+            .getTypeTable()
+            .getAndSaveNicknameFor(methodContext.getMethod().getInputType());
+
+    return GapicSurfaceTestCaseView.newBuilder()
+        .name(methodContext.getNamer().getTestCaseName(methodContext.getMethod()))
+        .surfaceMethodName(methodContext.getMethod().getSimpleName())
```

Comment:
Never use string directly from the model - go through the surface namer. In particular, this has the wrong casing (it's upper camel and needs to be lower camel)